hw/dv/dpi/usbdpi/usb_monitor.c

```diff
@@ -247,17 +318,19 @@ void monitor_usb(void *mon_void, FILE *mon_file, int loglevel, int tick,
   }
   switch (mon->state) {
     case MS_GET_PID:
-      if (((mon->bits & 0xf0) >> 4) ^ (mon->bits & 0x0f)) {
+      if (((mon->bits ^ 0xf0) >> 4) ^ (mon->bits & 0x0f)) {
```

Comment:
Can you put a comment in here on what this does? Also, you have changed the behavior here, was the previous behavior the wrong way around?

Reply:
Yes, I guess a comment is in order; the PID checking here was incomplete before; less strict than it should be. The upper and lower nibbles are supposed to be complements of each other. This is part of the error checking on the USB.